File: CS/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Drawing;

namespace SchedulerCustomDrawCurrentTimeMarker {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
            schedulerControl1.DayView.TimeRulers[0].ShowCurrentTime = false;
        }

        private void schedulerControl1_CustomDrawDayViewTimeRuler(object sender, DevExpress.XtraScheduler.CustomDrawObjectEventArgs e) {
            SchedulerControl scheduler = (SchedulerControl)sender;
            DateTime start = scheduler.Start;
            TimeSpan now = DateTime.Now.TimeOfDay;
            TimeRulerViewInfo viewInfo = (TimeRulerViewInfo)e.ObjectInfo;
            Rectangle rect = viewInfo.ContentBounds;
            Rectangle nowRect = rect;
            DayViewRowCollection visibleRows = scheduler.DayView.ViewInfo.VisibleRows;

            if (visibleRows.Count == 0)
                return;
            
            float offsetRatio = (now - scheduler.DayView.TopRowTime).Ticks / (float)TimeSpan.FromTicks(visibleRows[0].Interval.Duration.Ticks * visibleRows.Count).Ticks;

            if (offsetRatio < 0)
                return;

            nowRect.Height = 4;
            nowRect.Y += (int)Math.Round(rect.Height * offsetRatio - (float)nowRect.Height / 2);

            e.DrawDefault();
            e.Cache.FillRectangle(Brushes.Red, nowRect);
            e.Handled = true;
        }
    }
}

File: VB/Form1.vb
Imports System
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Drawing

Namespace SchedulerCustomDrawCurrentTimeMarker
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
            schedulerControl1.DayView.TimeRulers(0).ShowCurrentTime = False
        End Sub

        Private Sub schedulerControl1_CustomDrawDayViewTimeRuler(ByVal sender As Object, ByVal e As DevExpress.XtraScheduler.CustomDrawObjectEventArgs) Handles schedulerControl1.CustomDrawDayViewTimeRuler
            Dim scheduler As SchedulerControl = DirectCast(sender, SchedulerControl)
            Dim start As Date = scheduler.Start
            Dim now As TimeSpan = Date.Now.TimeOfDay
            Dim viewInfo As TimeRulerViewInfo = CType(e.ObjectInfo, TimeRulerViewInfo)
            Dim rect As Rectangle = viewInfo.ContentBounds
            Dim nowRect As Rectangle = rect
            Dim visibleRows As DayViewRowCollection = scheduler.DayView.ViewInfo.VisibleRows

            If visibleRows.Count = 0 Then
                Return
            End If

            Dim offsetRatio As Single = (now - scheduler.DayView.TopRowTime).Ticks / CSng(TimeSpan.FromTicks(visibleRows(0).Interval.Duration.Ticks * visibleRows.Count).Ticks)

            If offsetRatio < 0 Then
                Return
            End If

            nowRect.Height = 4
            nowRect.Y += CInt((Math.Round(rect.Height * offsetRatio - CSng(nowRect.Height) / 2)))

            e.DrawDefault()
            e.Cache.FillRectangle(Brushes.Red, nowRect)
            e.Handled = True
        End Sub
    End Class
End Namespace